Cabal/Distribution/PackageDescription/Parsec.hs

```diff
@@ -0,0 +1,364 @@
+          ++ "' must use section syntax. See the Cabal user guide for details."
+      where
+        displaySpecVersion (Left version)       = display version
+        displaySpecVersion (Right versionRange) =
+          case asVersionIntervals versionRange of
+            [] {- impossible -}           -> display versionRange
+            ((LowerBound version _, _):_) -> display (orLaterVersion version)
+
+    maybeWarnCabalVersion _ _ = return ()
+-}
+
+    handleFutureVersionParseFailure :: Version -> ParseResult a -> ParseResult GenericPackageDescription
+    handleFutureVersionParseFailure cabalVersionNeeded parseBody =
+        error "handleFutureVersionParseFailure"
+
+ {-
+      undefined (unless versionOk (warning message) >> parseBody)
+        `catchParseError` \parseError -> case parseError of
+        TabsError _   -> parseFail parseError
+        _ | versionOk -> parseFail parseError
+          | otherwise -> fail message
+      where versionOk = cabalVersionNeeded <= cabalVersion
+            message   = "This package requires at least Cabal version "
+                     ++ display cabalVersionNeeded
+    -}
+
+    checkForUndefinedFlags
+        :: GenericPackageDescription
+        -> ParseResult ()
+    checkForUndefinedFlags gpd = pure ()
+{-
+        let definedFlags = map flagName flags
+        mapM_ (checkCondTreeFlags definedFlags) (maybeToList mlib)
+        mapM_ (checkCondTreeFlags definedFlags . snd) sub_libs
+        mapM_ (checkCondTreeFlags definedFlags . snd) exes
+        mapM_ (checkCondTreeFlags definedFlags . snd) tests
+
+    checkCondTreeFlags :: [FlagName] -> CondTree ConfVar c a -> PM ()
+    checkCondTreeFlags definedFlags ct = do
+        let fv = nub $ freeVars ct
+        unless (all (`elem` definedFlags) fv) $
+            fail $ "These flags are used without having been defined: "
+                ++ intercalate ", " [ n | FlagName n <- fv \\ definedFlags ]
+-}
+
+parseConfVar :: [SectionArg Position] -> ParseResult ConfVar
+parseConfVar = error "parseConfVar"
+
+-- | Parse a list of fields, given a list of field descriptions,
+--   a structure to accumulate the parsed fields, and a function
+--   that can decide what to do with fields which don't match any
+--   of the field descriptions.
+parseFields :: [FieldDescr a]      -- ^ descriptions of fields we know how to
+                                   --   parse
+            -> UnrecFieldParser a  -- ^ possibly do something with
+                                   --   unrecognized fields
+            -> a                   -- ^ accumulator
+            -> [Field Position]    -- ^ fields to be parsed
+            -> ParseResult a
+parseFields descrs unrec ini fields = return ini -- TODO!
+
+-- TODO: build-depends is a special case
+-- TODO: add post check
+parseCondTree
+    :: forall a c.
+       [FieldDescr a]      -- ^ Field descriptions
+    -> UnrecFieldParser a  -- ^ How to parse unknown fields
```

Comment:
Oh if it's not propagated, then there's no point. Never mind! (I just kept seeing it in the signatures)